mods/gen2/scripts.js

```diff
@@ -117,6 +117,74 @@ exports.BattleScripts = {
 		this.singleEvent('AfterMove', move, null, pokemon, target, move);
 		if (!move.selfSwitch && target.hp > 0) this.runEvent('AfterMoveSelf', pokemon, target, move);
 	},
+	tryMoveHit: function (target, pokemon, move, spreadHit) {
+		let boostTable = [1, 4 / 3, 5 / 3, 2, 7 / 3, 8 / 3, 3];
+		let doSelfDestruct = true;
+		let damage = 0;
+
+		// First, check if the Pokémon is immune to this move.
+		if (move.ignoreImmunity !== true && !move.ignoreImmunity[move.type] && !target.runImmunity(move.type, true)) {
+			if (move.selfdestruct) {
+				this.faint(pokemon, pokemon, move);
+			}
+			return false;
+		}
+
+		// Now, let's calculate the accuracy.
+		let accuracy = move.accuracy;
+
+		// Calculate true accuracy for gen 2, which uses 0-255.
+		if (accuracy !== true) {
+			accuracy = Math.floor(accuracy * 255 / 100);
+			// Check also for accuracy modifiers.
+			if (!move.ignoreAccuracy) {
+				if (pokemon.boosts.accuracy > 0) {
+					accuracy *= boostTable[pokemon.boosts.accuracy];
+				} else {
+					accuracy = Math.floor(accuracy / boostTable[-pokemon.boosts.accuracy]);
+				}
+			}
+			if (!move.ignoreEvasion) {
+				if (target.boosts.evasion > 0 && !move.ignorePositiveEvasion) {
+					accuracy = Math.floor(accuracy / boostTable[target.boosts.evasion]);
+				} else if (target.boosts.evasion < 0) {
+					accuracy *= boostTable[-target.boosts.evasion];
+				}
+			}
+			accuracy = Math.min(accuracy, 255);
+			this.runEvent('ModifyAccuracy', target, pokemon, move, accuracy);
+		}
+		accuracy = this.runEvent('Accuracy', target, pokemon, move, accuracy);
+
+		if (accuracy !== true && this.random(255) >= accuracy) {
+			this.attrLastMove('[miss]');
+			this.add('-miss', pokemon);
+			damage = false;
+		}
+
+
+		if (move.category !== 'Status') {
+			// FIXME: The stored damage should be calculated ignoring Substitute.
+			// https://github.com/Zarel/Pokemon-Showdown/issues/2598
+			target.gotAttacked(move, damage, pokemon);
+		}
+
+		
+		if (move.selfdestruct && doSelfDestruct) {
+			this.faint(pokemon, pokemon, move);
+		}
+
+
+
+		if (move.ohko) this.add('-ohko');
```

Comment:
Isn't One Hit KO supposed to be abbreviated in caps?

Reply:
only for grammar purposes. It's stored as `ohko` in PS iirc